driver-core/src/main/com/mongodb/internal/TimeoutSettings.java

```diff
@@ -37,35 +37,38 @@ public class TimeoutSettings {
     @Nullable
     private final Long defaultTimeoutMS;
 
-    private final long maxAwaitTimeMS;
-
-    // Deprecated timeouts
-    private final long readTimeoutMS;
+    // Deprecated configuration timeout options
+    private final long readTimeoutMS; // aka socketTimeoutMS
+    private final long maxWaitTimeMS; // aka waitQueueTimeoutMS
+    @Nullable
+    private final Long wTimeoutMS;
 
+    // Deprecated options for CRUD methods
     private final long maxTimeMS;
-
+    private final long maxAwaitTimeMS;
     private final long maxCommitTimeMS;
-    @Nullable
-    private final Long wTimeoutMS;
 
     public static final TimeoutSettings DEFAULT = create(MongoClientSettings.builder().build());
 
     public static TimeoutSettings create(final MongoClientSettings settings) {
-        return new TimeoutSettings(settings.getClusterSettings().getServerSelectionTimeout(TimeUnit.MILLISECONDS),
+        return new TimeoutSettings(
+                settings.getClusterSettings().getServerSelectionTimeout(TimeUnit.MILLISECONDS),
                 settings.getSocketSettings().getConnectTimeout(TimeUnit.MILLISECONDS),
                 settings.getSocketSettings().getReadTimeout(TimeUnit.MILLISECONDS),
-                settings.getTimeout(TimeUnit.MILLISECONDS));
+                settings.getTimeout(TimeUnit.MILLISECONDS),
+                settings.getConnectionPoolSettings().getMaxWaitTime(TimeUnit.MILLISECONDS));
     }
 
     public TimeoutSettings(
-            final long serverSelectionTimeoutMS, final long connectTimeoutMS, final long readTimeoutMS, @Nullable final Long timeoutMS) {
-        this(timeoutMS, null, serverSelectionTimeoutMS, connectTimeoutMS, readTimeoutMS, 0, 0, 0, null);
+            final long serverSelectionTimeoutMS, final long connectTimeoutMS, final long readTimeoutMS,
+            @Nullable final Long timeoutMS, final long maxWaitTimeMS) {
+        this(timeoutMS, null, serverSelectionTimeoutMS, connectTimeoutMS, readTimeoutMS, 0, 0, 0, null, maxWaitTimeMS);
```

Comment:
I added maxWaitTimeMS, in a way that seemed consistent with readTimeoutMS, but I was also not sure whether we intended to pass these in, since both were deprecated. Should these really be passed in, or should they be set using one of the "with" methods below?

Reply:
Its needed in the constructor, so it can be copied via the with methods.
We should eventually make this constructor private or annotate visibility test only.